homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,179 @@
+
+
+class SwitcherControl(SwitchDevice):
+    """Home Assistant switch entity."""
+
+    def __init__(self, device_data: Any) -> None:
+        """Initialize the entity."""
+        # pylint: disable=unused-import
+        from aioswitcher.devices import SwitcherV2Device
+        # pylint: enable=unused-import
+        self._self_initiated = False
+        self._device_data = device_data  # type: SwitcherV2Device
+        self._state = device_data.state
+        self._unsub_dispatcher = None  # type: Optional[Callable[..., Any]]
+
+    @property
+    def electric_current(self) -> float:
+        """Return the electric current."""
+        return cast(float, self._device_data.electric_current)
+
+    @property
+    def remaining_time(self) -> str:
+        """Return the remaining time to off command."""
+        return cast(str, self._device_data.remaining_time)
+
+    @property
+    def auto_off_set(self) -> str:
+        """Return the auto off configuration set."""
+        return cast(str, self._device_data.auto_off_set)
+
+    @property
+    def device_name(self) -> str:
+        """Return the device's name."""
+        return cast(str, self._device_data.name)
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False, entity pushes its state to HA."""
+        return False
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "{}-{}".format(
+            self._device_data.device_id, self._device_data.mac_addr)
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        from aioswitcher.consts import STATE_ON as SWITCHER_STATE_ON
+        return cast(bool, self._state == SWITCHER_STATE_ON)
+
+    @property
+    def current_power_w(self) -> int:
+        """Return the current power usage in W."""
+        return cast(int, self._device_data.power_consumption)
+
+    @property
+    def device_state_attributes(self) -> Dict:
+        """Return the optional state attributes."""
+        from aioswitcher.consts import WAITING_TEXT
+
+        attribs = {}
+
+        for prop, attr in PROPERTIES_TO_DEVICE_ATTRIBUTES.items():
+            value = getattr(self, prop)
+            if value and value is not WAITING_TEXT:
+                attribs[attr] = value
+
+        return attribs
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        from aioswitcher.consts import (STATE_OFF as SWITCHER_STATE_OFF,
+                                        STATE_ON as SWITCHER_STATE_ON)
+        return self._state in [SWITCHER_STATE_ON, SWITCHER_STATE_OFF]
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        if not self.hass:
+            return
+        self._unsub_dispatcher = async_dispatcher_connect(
+            self.hass, SIGNAL_SWITCHER_DEVICE_UPDATE, self.async_update_data)
+
+    async def async_will_remove_from_hass(self) -> None:
```

Comment:
This will never be called since the component isn't using config entries -- can be removed.

Reply:
@andrewsayre 
Sorry, I missed that one.
I'm thinking of replacing it with an event listener for the `EVENT_HOMEASSISTANT_STOP`.
What do you think? Is this perhaps redundant?

Reply:
It's not necessary and won't do anything.  The component doesn't support unloading, so there's no need to implement this functionality.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/e433319eefd2f2373b3ac5b8752371beade902fc).